### ISSUE TITLE
parsing failure with nonatomic stored procedures and Redshift dialect

### ISSUE THREAD

**Issue by ramsespostnl** [sqlfluff/sqlfluff#7097]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The `nonatomic` Redshift stored procedure keyword is not recognized by the sqlfluff parser.

### Expected Behaviour

Since this is valid Redshift SQL, sqlfluff should be able to parse it.

### Observed Behaviour

Parsing any stored procedure with nonatomic keyword will fail. Also linting or formatting fails due to the parsing error.  In the 'How to reproduce' I've written the parsing example. 

### How to reproduce

When parsing this example file with `sqlfluff parse path/to/file.sql` we get an error.

```sql
create or replace procedure etl.test_sp_na ()
nonatomic
as
$$
begin
    select 1;
end;
$$
language plpgsql
security invoker;
```

Error:

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Delimited: [<Ref: 'StatementSegment'>]>"
[L:  1, P:  1]      |        word:                                                 'create'
[L:  1, P:  7]      |        whitespace:                                           ' '
[L:  1, P:  8]      |        word:                                                 'or'
[L:  1, P: 10]      |        whitespace:                                           ' '
[L:  1, P: 11]      |        word:                                                 'replace'
[L:  1, P: 18]      |        whitespace:                                           ' '
[L:  1, P: 19]      |        word:                                                 'procedure'
[L:  1, P: 28]      |        whitespace:                                           ' '
[L:  1, P: 29]      |        word:                                                 'etl'
[L:  1, P: 32]      |        dot:                                                  '.'
[L:  1, P: 33]      |        word:                                                 'test_sp_na'
[L:  1, P: 43]      |        whitespace:                                           ' '
[L:  1, P: 44]      |        start_bracket:                                        '('
[L:  1, P: 45]      |        end_bracket:                                          ')'
[L:  1, P: 46]      |        newline:                                              '\n'
[L:  2, P:  1]      |        word:                                                 'nonatomic'
[L:  2, P: 10]      |        newline:                                              '\n'
[L:  3, P:  1]      |        word:                                                 'as'
[L:  3, P:  3]      |        newline:                                              '\n'
[L:  4, P:  1]      |        dollar_quote:                                         '$$\nbegin\n    select 1;\nend;\n$$'
[L:  8, P:  3]      |        newline:                                              '\n'
[L:  9, P:  1]      |        word:                                                 'language'
[L:  9, P:  9]      |        whitespace:                                           ' '
[L:  9, P: 10]      |        word:                                                 'plpgsql'
[L:  9, P: 17]      |        newline:                                              '\n'
[L: 10, P:  1]      |        word:                                                 'security'
[L: 10, P:  9]      |        whitespace:                                           ' '
[L: 10, P: 10]      |        word:                                                 'invoker'
[L: 10, P: 17]      |        semicolon:                                            ';'
[L: 10, P: 18]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'create or replace
                       | procedure etl.test_sp_...'
```

If I only remove the nonatomic keyword then, this can be parsed.

### Dialect

redshift

### Version

`sqlfluff, version 3.4.2`

### Configuration

`.sqlfluff`:

```
[sqlfluff]
dialect = redshift
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by franloza** [comment]
I created a PR to fix this: https://github.com/sqlfluff/sqlfluff/pull/7123
